test/cluster/object_store/test_backup.py

```
@@ -1190,6 +1193,194 @@ async def test_restore_tablets_with_different_tablet_hints(build_mode: str, mana
+
+        snap_name, _ = await take_snapshot(src_ks, servers, manager, logger)
+        await asyncio.gather(*(do_backup(s, snap_name, f'{s.server_id}/{snap_name}', src_ks, 'cf1', object_storage, manager, logger) for s in servers))
+
+        async with new_test_keyspace(manager, f"WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': {topology.rf}}}") as dst_ks:
+            await cql.run_async(f"CREATE TABLE {dst_ks}.cf2 {dst_schema} WITH tablets = {{'min_tablet_count': {tablet_count}}};")
+
+            manifests = [ f'{s.server_id}/{snap_name}/manifest.json' for s in servers ]
+            logger.info(f'Restore {src_ks}.cf1 into {dst_ks}.cf2 declared as {dst_schema}')
+            tid = await manager.api.restore_tablets(servers[1].ip_addr, dst_ks, 'cf2', snap_name, servers[0].datacenter, object_storage.address, object_storage.bucket_name, manifests)
+            status = await manager.api.wait_task(servers[1].ip_addr, tid)
+            logger.info(f'Restore into an incompatible {dst_ks}.cf2 reported state={status["state"] if status else None}')
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)